tracing-effectful/tracing-effectful.cabal

```diff
@@ -0,0 +1,114 @@
+cabal-version:      3.0
+-- The cabal-version field refers to the version of the .cabal specification,
+-- and can be different from the cabal-install (the tool) version and the
+-- Cabal (the library) version you are using. As such, the Cabal (the library)
+-- version used must be equal or greater than the version stated in this field.
+-- Starting from the specification version 2.2, the cabal-version field must be
+-- the first thing in the cabal file.
+
+-- Initial package description 'tracing-effectful' generated by
+-- 'cabal init'. For further documentation, see:
+--   http://haskell.org/cabal/users-guide/
+--
+-- The name of the package.
+name:               tracing-effectful
+
+-- The package version.
+-- See the Haskell package versioning policy (PVP) for standards
+-- guiding when and how versions should be incremented.
+-- https://pvp.haskell.org
+-- PVP summary:     +-+------- breaking API changes
+--                  | | +----- non-breaking API additions
+--                  | | | +--- code changes with no API change
+version:            0.1.0.0
+
+-- A short (one-line) description of the package.
+-- synopsis:
+
+-- A longer description of the package.
+-- description:
+
+-- The license under which the package is released.
+license:            BSD-3-Clause
+
+-- The file containing the license text.
+license-file:       LICENSE
+
+-- The package author(s).
+author:             Adrien Duclos
+
+-- An email address to which users can send suggestions, bug reports, and patches.
+maintainer:         adrien.duclos@scrive.com
+
+-- A copyright notice.
+-- copyright:
+category:           Development
+build-type:         Simple
+
+-- Extra doc files to be distributed with the package, such as a CHANGELOG or a README.
+extra-doc-files:    CHANGELOG.md
+
+-- Extra source files to be distributed with the package, such as examples, or a tutorial module.
+-- extra-source-files:
+
+common language
+    ghc-options: -Wall -Wcompat -Wno-unticked-promoted-constructors
+
+    default-language:   Haskell2010
+
+    default-extensions: DataKinds
+                        FlexibleContexts
+                        KindSignatures
+                        LambdaCase
+                        TypeFamilies
+                        TypeOperators
+
+
+library
+    -- Import common warning flags.
+    import:           language
+
+    -- Modules exported by the library.
+    exposed-modules:  Effectful.Tracing
+
+    -- Modules included in this library but not exported.
+    -- other-modules:
+
+    -- LANGUAGE extensions used by modules in this package.
+    -- other-extensions:
+
+    -- Other library packages from which modules are imported.
+    build-depends:   base ^>=4.17.2.1
```

Comment:
This makes it work with GHC 9.4. Also, please put upper bound for `base` to be `5` so we don't have to bump it when updating GHC.